### PR TITLE
Rspec setup

### PR DESCRIPTION
**Pull request by lucamarial**
[PT link](https://www.pivotaltracker.com/story/show/169253707)

## This PR contains
In this PR we are installing RSpec for unit testing.

## PR content
- RSpec
- Shoulda Matchers
- FactoryBot

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -0,0 +1 @@
+--require spec_helper
```
Comment (tochman) [inline .rspec]: please change to `rails_helper`

File: Gemfile [hunk 2]
```diff
@@ -12,6 +12,9 @@ gem 'rack-cors'
 group :development, :test do
   gem 'pry-rails'
   gem 'pry-byebug'
+  gem 'rspec-rails'
+  gem 'shoulda-matchers'
```
Comment (tochman) [inline Gemfile:16]: You include this gem but you don't use it?

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  #
+  # The different available types are documented in the features, such as in
+  # https://relishapp.com/rspec/rspec-rails/docs
+  config.infer_spec_type_from_file_location!
+
+  # Filter lines from Rails gems in backtraces.
+  config.filter_rails_from_backtrace!
+  # arbitrary gems may also be filtered via:
+  # config.filter_gems_from_backtrace("gem name")
+end
```
Comment (tochman) [inline spec/rails_helper.rb:26]: Please clean up this file

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  #     --seed 1234
+  config.order = :random
+
+  # Seed global randomization in this process using the `--seed` CLI option.
+  # Setting this allows you to use `--seed` to deterministically reproduce
+  # test failures related to randomization by passing the same `--seed` value
+  # as the one that triggered the failure.
+  Kernel.srand config.seed
+=end
+end
```
Comment (tochman) [inline spec/spec_helper.rb]: Please clean up this file

**Review by yasmineezequiel — APPROVED** [review]

File: spec/support/factory_bot.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
```
Comment (tochman) [inline spec/support/factory_bot.rb]: please move this configuration to `rails_helper` and delete this file

File: spec/support/shoulda_matcher.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+Shoulda::Matchers.configure do |config|
+  config.integrate do |with|
+    with.test_framework :rspec
+    with.library :rails
+  end
+end
+
+RSpec.configure do |config|
+  config.include(Shoulda::Matchers::ActiveRecord, type: :model)
+end
```
Comment (tochman) [inline spec/support/shoulda_matcher.rb]: please move this configuration to `rails_helper` and delete this file

File: spec/support/shoulda_matcher.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+Shoulda::Matchers.configure do |config|
+  config.integrate do |with|
+    with.test_framework :rspec
+    with.library :rails
+  end
+end
```
Comment (tochman) [inline spec/support/shoulda_matcher.rb]: please move this to `rails_helper`

**Review by tochman — CHANGES_REQUESTED** [review]

**Review by shsteindorsson — APPROVED** [review]

**Review by beccaburns — APPROVED** [review]

**Review by DIzosimov — APPROVED** [review]

**Review by shsteindorsson — APPROVED** [review]

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**